letsencrypt/cli.py

```diff
@@ -733,6 +732,32 @@ def install(config, plugins):
     le_client.enhance_config(domains, config)
 
 
+def _set_by_cli(variable):
+    """
+    Return True if a particular config variable has been set by the user
+    (CLI or config file) including if the user explicitly set it to the
+    default.  Returns False if the variable was assigned a default variable.
+    """
+    if _set_by_cli.detector is None:
+        # Setup on first run: `detector` is a weird version of config in which
+        # the default value of every attribute is wrangled to be boolean-false
+        plugins = plugins_disco.PluginsRegistry.find_all()
+        # reconstructed_args == sys.argv[1:], or whatever was passed to main()
+        reconstructed_args = _parser.args + [_parser.verb]
+        default_args = prepare_and_parse_args(plugins, reconstructed_args, empty_defaults=True)
+        _set_by_cli.detector = configuration.NamespaceConfig(default_args, fake=True)
+    try:
+        # Is detector.variable something that isn't false?
+        if _set_by_cli.detector.__getattr__(variable):
```

Comment:
Will this work for arguments that we're setting to false, such as --no-hsts or --no-uir?